content/en/blog/_posts/2022-12-6-cross-namespace-data-sources-alpha.md

```diff
@@ -0,0 +1,137 @@
+
+## Trying it out
+
+The following things are required to use cross namespace volume provisioning:
+
+* Enable the `AnyVolumeDataSource` and `CrossNamespaceVolumeDataSource` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) for the kube-apiserver, kube-controller-manager
+* Install a CRD for the specific `VolumeSnapShot` controller
+* Install the CSI Provisioner controller and enable the `CrossNamespaceVolumeDataSource` feature gate
+* Install the CSI driver
+* Install a CRD for ReferenceGrants
+
+## Putting it all together
+
+To see how this works, you can install the sample and try it out.
+This sample do to create PVC in dev namespace from VolumeSnapshot in prod namespace.
+That is a simple example. For real world use, you might want to use a more complex approach.
+
+### Prerequisites
+
+* Kubernetes cluster is being deployed with `AnyVolumeDataSource` and `CrossNamespaceVolumeDataSource` features enabled
+* There are two namespaces, dev and prod
+* CSI driver is being deployed
+* There is an acquired VolumeSnapShot named of "new-snapshot-demo" on prod
+* ReferenceGrants CRD is being deployed
+
+### Add referencegrants permission to CSI Provisioner RBAC
+
+Access to referencegrants is only needed when the CSI driver
+has the `CrossNamespaceVolumeDataSource` controller capability.
+Therefore, external-provisioner adds "get", "list", "watch"
+permissions for "referencegrants" on "gateway.networking.k8s.io".
+
+```yaml
+  - apiGroups: ["gateway.networking.k8s.io"]
+    resources: ["referencegrants"]
+    verbs: ["get", "list", "watch"]
+```
+
+### The CSI Provisioner controller and enable the CrossNamespaceVolumeDataSource feature gate
+
+Add `--feature-gates=CrossNamespaceVolumeDataSource=true` to csi-provisioner container
+
+```yaml
+      - args:
+        - -v=5
+        - --csi-address=/csi/csi.sock
+        - --feature-gates=Topology=true
+        - --feature-gates=CrossNamespaceVolumeDataSource=true
+        image: csi-provisioner:latest
+        imagePullPolicy: IfNotPresent
+        name: csi-provisioner
+```
+
+### Create a ReferenceGrant. Here's a manifest for an example ReferenceGrant.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: ReferenceGrant
+metadata:
+  name: allow-prod-pvc
+  namespace: prod
+spec:
+  from:
+  - group: ""
+    kind: PersistentVolumeClaim
+    namespace: dev
+  to:
+  - group: snapshot.storage.k8s.io
+    kind: VolumeSnapshot
+    name: new-snapshot-demo
+```
+
+### Create a PersistentVolumeClaim by using cross namespace data source
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: example-pvc
+  namespace: dev
+spec:
+  storageClassName: csi-hostpath-sc
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+  dataSourceRef:
+    apiGroup: snapshot.storage.k8s.io
+    kind: VolumeSnapshot
+    name: new-snapshot-demo
+    namespace: prod
+  volumeMode: Filesystem
+```
+
```

Comment:
Any summary here?

For example: “Kubernetes creates a PersistentVolume and the CSI driver populates it with data from the snapshot”.